src/main/java/meteordevelopment/meteorclient/systems/modules/render/Trajectories.java

```diff
@@ -192,16 +200,30 @@ public void clear() {
         public void calculate() {
             addPoint();
 
-            while (true) {
-                HitResult result = simulator.tick();
-
-                if (result != null) {
-                    processHitResult(result);
-                    break;
+            if (simulationSteps.get() == 0) {
+                while (true) {
+                    HitResult result = simulator.tick();
+    
+                    if (result != null) {
+                        processHitResult(result);
+                        break;
+                    }
+    
+                    addPoint();
+                }
+            } else {
+                for (int i = 0; i < simulationSteps.get(); i++) {
```

Comment:
Same as before. You could merge the two loops by using `Integer.MAX_VALUE` as the limit if `simulationSteps` is 0.